### DISCUSSION TITLE
Globe Clipping Plane

### DISCUSSION THREAD

**Discussion by KorecGroup-AliJafri** [zouyaoji/vue-cesium#562] [Q&A] [answered]
Hi, what's the best way to implement a clipping plane on a globe? When I try to define a clipping plane via GeoJson and use the polygon defined, the vue-cesium implementation gives me an error of `ClippingPolygonCollection` and `ClippingPolygon` not being a constructor.

**Comment by zouyaoji** [comment]
https://github.com/zouyaoji/vue-cesium/discussions/485

**Reply by zouyaoji** [reply, marked as answer]
If you want to bind: clippingPlanes="clippingPlanes", you need to first define a const clippingPlanes = ref(null); and then initialize clippingPlanes in the @ready event of vc-viewer. If you need to use Cesium to initialize, you must write it like this. Unless you put the introduction of Cesium.js into index.html.

You can also write this in the ready event of the vc-primitive-tileset component:
tileset._clippingPlanes = new Cesium.ClippingPlaneCollection({
planes: [new Cesium.ClippingPlane(new Cesium.Cartesian3(0.0, 0.0, -1.0), 0.0)],
edgeWidth: 1.0
})

**Reply by KorecGroup-AliJafri** [reply]
Thank you! It works great!
